### PR TITLE
Adding Charon keyboard to Charue family

### PR DESCRIPTION
**Pull request by PiKeeb**
## Description

Added Charon keyboard to the Charue family

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (**addition** or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/charue/charon/info.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{
+    "keyboard_name": "Charon",
+    "url": "https://charue-design.com/products/charon",
+    "maintainer": "ILWrites",
+    "width": 3,
+    "height": 2,
```
Comment (drashna) [inline keyboards/charue/charon/info.json:6]:
```suggestion
```

File: keyboards/charue/charon/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/charue/charon/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by PiKeeb** [comment]
@drashna done. All requested changes are in!

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
Please refer to the QMK [PR Checklist](https://docs.qmk.fm/#/pr_checklist) and work through all the required changes.
There are a few obvious things in the PR (most obvious being `QMKBEST`) that show that the PR checklist hasn't been consulted; this document helps us keep things standard as much as possible and makes it easier for reviews to occur.

File: keyboards/charue/charon/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/charue/charon/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/charue/charon/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Charon
+
+![Charon](https://i.imgur.com/fnQLZ2T.jpg)
```
Comment (fauxpark) [inline keyboards/charue/charon/readme.md:3]:
```suggestion
![Charon](https://i.imgur.com/fnQLZ2Th.jpg)
```

File: keyboards/charue/charon/readme.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: [Charon PCB](https://charue-design.com/collections/charon/)
+* Hardware Availability: [Charue Design](https://charue-design.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make charon:default
+
+Flashing example for this keyboard:
+
+    make charon:default:flash
```
Comment (fauxpark) [inline keyboards/charue/charon/readme.md:17]:
```suggestion
    make charue/charon:default

Flashing example for this keyboard:

    make charue/charon:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by PiKeeb** [comment]
@tzarc

> Please refer to the QMK [PR Checklist](https://docs.qmk.fm/#/pr_checklist) and work through all the required changes.
> There are a few obvious things in the PR (most obvious being `QMKBEST`) that show that the PR checklist hasn't been consulted; this document helps us keep things standard as much as possible and makes it easier for reviews to occur.

Thanks for your comments. The `default` or `via` keymaps does not contain `QMKURL`/`QMKBEST`. It's in the `debug` keymap and is there for debug purposes (to check how it handles these `SEND_STRING` commands). I can change these to something different, if needed.

Before doing the PR, I've tried to follow the PR Checklist to the T, but I think I might have slipped on a few points. I'll check and adjust, but can you look at my PR as well? That would be very helpful.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**